src/functionalTest/java/uk/gov/hmcts/reform/locationrefdata/cameltest/LrdCourtVenueTest.java

```diff
@@ -0,0 +1,164 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import uk.gov.hmcts.reform.data.ingestion.configuration.AzureBlobConfig;
+import uk.gov.hmcts.reform.data.ingestion.configuration.BlobStorageCredentials;
+import uk.gov.hmcts.reform.locationrefdata.camel.binder.CourtVenue;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.LrdIntegrationBaseTest;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.SpringStarter;
+import uk.gov.hmcts.reform.locationrefdata.config.LrdCamelConfig;
+import uk.gov.hmcts.reform.locationrefdata.configuration.BatchConfig;
+
+import java.io.FileInputStream;
+import java.util.Date;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.javatuples.Triplet.with;
+import static org.springframework.util.ResourceUtils.getFile;
+import static uk.gov.hmcts.reform.data.ingestion.camel.util.MappingConstants.SCHEDULER_START_TIME;
+
+@TestPropertySource(properties = {"spring.config.location=classpath:application-integration.yml,"
+    + "classpath:application-leaf-integration.yml"})
+@CamelSpringBootTest
+@MockEndpoints("log:*")
+@ContextConfiguration(classes = {LrdCamelConfig.class, CamelTestContextBootstrapper.class,
+    JobLauncherTestUtils.class, BatchConfig.class, AzureBlobConfig.class, BlobStorageCredentials.class},
+    initializers = ConfigFileApplicationContextInitializer.class)
+@SpringBootTest
+@EnableAutoConfiguration(exclude = JpaRepositoriesAutoConfiguration.class)
+@EnableTransactionManagement
+@SqlConfig(dataSource = "dataSource", transactionManager = "txManager",
+    transactionMode = SqlConfig.TransactionMode.ISOLATED)
+@SuppressWarnings("unchecked")
+public class LrdCourtVenueTest extends LrdIntegrationBaseTest {
+
+    @Value("${start-route}")
+    private String startRoute;
+
+    @Value("${archival-route}")
+    String archivalRoute;
+
+    @Autowired
+    @Qualifier("springJdbcTransactionManager")
+    protected PlatformTransactionManager platformTransactionManager;
+
+    private static final String ROUTE_TO_EXECUTE = "lrd-court-venue-load";
+
+    @BeforeEach
+    public void init() {
+        SpringStarter.getInstance().restart();
+        camelContext.getGlobalOptions()
+            .put(SCHEDULER_START_TIME, String.valueOf(new Date(System.currentTimeMillis()).getTime()));
+    }
+
+    @Test
+    @Sql(scripts = {"/testData/truncate-lrd-court-venue.sql"})
+    void testTaskletSuccess() throws Exception {
+        testCourtVenueInsertion();
+    }
+
+    private void testCourtVenueInsertion() throws Exception {
+        setLrdCamelRouteToExecute(ROUTE_TO_EXECUTE);
```

Comment:
I think it is better to have these in `@BeforEach`. instead of having in each `@Test`